src/api/wallet/WalletApiImpl.ts

```diff
@@ -248,13 +247,14 @@ export class WalletApiImpl implements WalletApi {
       isConnected,
       userAddress: accounts[0],
       networkId: isConnected ? +chainId : undefined,
+      web3: this._web3,
     }
   }
 
   private async _notifyListeners(): Promise<void> {
-    await Promise.resolve()
+    const blockNumber = await this._blockNumber
```

Comment:
No need for parenthesis when calling `this._blockNumber`?

Reply:
Getter, so nope!

Reply:
Cool, good to know!